src/lib/Serialize.hs

```diff
@@ -63,13 +63,15 @@ getDexString (DataCon _ _ _ 0 [_, xs]) = case tryParseStringContent xs of
     tryParseStringContent tabAtom  = do
       TabLam (TabLamExpr i body) <- return tabAtom
       Fin (IdxRepVal n) <- return $ binderType i
-      Block _ (Nest castDecl (Nest offDecl Empty)) loadExpr <- return body
+      Abs (Nest castDecl (Nest offDecl Empty)) loadExpr <-
+        return $ inlineLastDeclBlock body
```

Comment:
There's an easy edit to this code that won't require us to go through `inlineLastDeclBlock` can you please try this:
```hs
Block _ (Nest castDecl (Nest offDecl (Nest loadDecl))) resultAtom <- return body
```

Then, just add a check that `resultAtom` is a `Var` that's bound by `loadDecl`

Reply:
Done.